src/test/datascience/notebook/helper.ts

```diff
@@ -245,7 +245,7 @@ export async function waitForKernelToGetSelected(kernelNameSearch: string) {
     const id = kernels?.find((k) => k.label.includes(kernelNameSearch))?.id;
 
     // Send a select kernel on the active notebook editor
-    void commands.executeCommand('notebook.selectKernel', { id });
+    void commands.executeCommand('notebook.selectKernel', { id, extension: JVSC_EXTENSION_ID });
```

Comment:
I don't think this is the right place to force the right kernel to get selected.
This forces the kernel to get selected.
We have tests that wait for VS Code to auto select the right kernel.
E.g. when we open a notebook with a java kernel, we would like the java kernel to get selected by VS Code.
This will always ensure that we forcefully select the kernel & if the code in the extension does't pre-select the right kernel, it doesn't matter as we force it in here.

Reply:
I think we should remove this line & move it close to where ever we need this.
All of our existings test pass without this today.

Reply:
Nope ignore my comments, the one I'm referring to is `waitForKernelToGetAutoSelected` missed `auto`. this is a new function.

Reply:
Shouldn't we rename this function `changeKernel..../switchKernel...`
As we aren't waiting for it to get selected, but we're forcing a new selection

Reply:
Sure